mathesar_ui/src/component-library/confirmation/Confirmation.svelte

```diff
@@ -0,0 +1,95 @@
+<script lang="ts">
+  import Button from '../button/Button.svelte';
+  import Icon from '../icon/Icon.svelte';
+  import Modal from '../modal/Modal.svelte';
+  import Spinner from '../spinner/Spinner.svelte';
+  import type { ConfirmationController } from './ConfirmationController';
+
+  export let controller: ConfirmationController;
+
+  $: ({ modal, confirmationProps, resolve } = controller);
+  $: ({
+    title,
+    body,
+    proceedButton,
+    cancelButton,
+    onProceed,
+    onSuccess,
+    onError,
+  } = $confirmationProps);
+  $: bodyParagraphs = Array.isArray(body) ? body : [body];
+
+  let isProcessing = false;
+
+  function handleCancelButton() {
+    $resolve(false);
+    modal.close();
+  }
+
+  async function handleProceedButton() {
+    try {
+      isProcessing = true;
+      await onProceed();
+      onSuccess();
+      $resolve(true);
+      modal.close();
+    } catch (error) {
+      onError(error);
+    } finally {
+      isProcessing = false;
+    }
+  }
+
+  function onClose() {
+    // When the user closes the confirmation modal by clicking on the "X" button
+    // we want to resolve the promise.
+    //
+    // In a "proceed" situation, we call `$resolve(true)` first and _then_ close
+    // the modal afterwards. Here, we're calling `deny()` no matter what
+    // happened, which means `deny()` we'll ultimately call `$resolve(true)`
+    // and then `$resolve(false)` which may seem a bit strange. This is fine
+    // because though because a Promise can only be resolved once.
+    $resolve(false);
```

Comment:
This is a hacky solution. While this would work perfectly fine, I'd rather have us think of better ways to handle this.

Having said that, I don't think we need to spend further time on this at the moment, and could safely put it off for later.